CONTRIBUTING.md

```diff
@@ -45,22 +45,134 @@ To add a new plugin you need to do the following.
+| **deprecateExtraOptions**  | Indicating whether to deprecate the `extraOptions` attribute. Mainly used for old plugins.                                                                                                                   | No       | `false`       |
+| **description**            | A brief description of the plugin. Can also be used for non-normative documentation, warnings, tips and tricks.                                                                                              | No       | `null`        |
+| **extraConfig**            | Additional configuration for the plugin.                                                                                                                                                                     | No       | `{}`          |
+| **extraOptions**           | Module options for the plugin, to be added _outside_ of the `settings` option. These should be Nixvim-specific options.                                                                                       | No       | `{}`          |
+| **extraPackages**          | Extra packages to include.                                                                                                                                                                                   | No       | `[]`          |
+| **extraPlugins**           | Extra plugins to include.                                                                                                                                                                                    | No       | `[]`          |
+| **hasConfigAttrs**         | Indicating whether the plugin has configuration attributes.                                                                                                                                                  | No       | `true`        |
+| **hasSettings**            | Indicating whether the plugin has settings. A `settings` option will be created if true.                                                                                                                     | No       | `true`        |
+| **imports**                | Additional modules to import.                                                                                                                                                                                | No       | `[]`          |
+| **installPackage**         | Whether to install the package. This is a very niche option that can be used when the final package overrides the plugin's configuration. In most cases, it shouldn't disabled.                              | No       | `true`        |
```

Comment:
I don't think it's accurate to say the _package_ overrides the _plugin_ config...

This wording should be a bit better, but still hard to git it exactly right:

```suggestion
| **installPackage**         | Whether to install the package. This is a very niche option, but one use case is adding additional overrides to the final package. In most cases, it shouldn't disabled.                                        | No       | `true`        |
```

---

The fact this is so clunky to describe makes me wonder if we should have a transformer argument instead of a toggle :thinking: 

Something like `packageTransformer ? (pkg: pkg)` "Used to apply final transformations to the configured package, such as necessary overrides"

Maybe I'll open a PR once this is merged

Reply:
I think the transformer would be a good idea depending on how this implemented and presented with arguments. The `installPackage` is self explanatory, but I still don't know how to rephrase it outside of `Just install the package or not`. It mostly depends on side args in the same func. Maybe we could just stick to a very simple `Should the package be installed or not` for now, and then try to find a better way to customize the plugin via, as you said for example, a transformer.

Reply:
It's a deal then 🤝!

Reply:
done!

Reply:
I've opened #2524 to remove the option. I think regardless of the outcome of that PR we probably don't need this one documented.


```suggestion
```